source/simulator/assembly.cc

```diff
@@ -1701,6 +1807,839 @@ namespace aspect
+                               * time_step
+                               * (current_u
+                                  * scratch.face_finite_element_values.normal_vector(q))
+                               * fixed_temperature_value
+                               * scratch.face_phi_field[i]
+                               :
+                               0.)
+                           )
+                           *
+                           scratch.face_finite_element_values.JxW(q);
+
+                        // local_matrix terms
+                        for (unsigned int j=0; j<advection_dofs_per_cell; ++j)
+                          {
+                            data.local_matrix(i,j)
+                            += (- time_step *  conductivity
+                                * scratch.face_grad_phi_field[i]
+                                * scratch.face_finite_element_values.normal_vector(q)
+                                * scratch.face_phi_field[j]
+
+                                - time_step *  conductivity
+                                * scratch.face_grad_phi_field[j]
+                                * scratch.face_finite_element_values.normal_vector(q)
+                                * scratch.face_phi_field[i]
+
+                                + time_step
+                                * (density_c_P + latent_heat_LHS)
+                                * penalty
+                                * scratch.face_phi_field[i]
+                                * scratch.face_phi_field[j]
+
+                                + (current_u * scratch.face_finite_element_values.normal_vector(q) < 0
+                                   ?
+                                   - (density_c_P + latent_heat_LHS)
+                                   * time_step
+                                   * (current_u
+                                      * scratch.face_finite_element_values.normal_vector(q))
+                                   * scratch.face_phi_field[i]
+                                   * scratch.face_phi_field[j]
+                                   :
+                                   0.)
+
+                               )
+                               * scratch.face_finite_element_values.JxW(q);
+
+                          }
+                      }
+                  }
+              }
+            else
+              {
+                //Neumann temperature term - no non-zero contribution as only homogeneous Neumann boundary conditions are implemented elsewhere for temperature
+              }
+          }
+        else
+          {
+            //interior face - no contribution on RHS
+            Assert (cell->neighbor(f).state() == IteratorState::valid,
+                    ExcInternalError());
+            typename DoFHandler<dim>::cell_iterator neighbor = cell->neighbor(f); //note: NOT active_cell_iterator, so this includes cells that are refined.
```

Comment:
Can this be made `const`? This is mostly to avoid the need for readers (like me) to think about where you might change this afterwards.